core/src/lib/parse_conf.h

```diff
@@ -475,15 +475,28 @@ void IndentConfigItem(PoolMem& cfg_str,
                       const char* config_item,
                       bool inherited = false);
 
+/* this function is used as an initializer in foreach_res, so we can null
+ * the pointer passed into and also get a reference to the configuration that
+ * we then keep for the lifetime of the loop.
+ */
+inline std::shared_ptr<ConfigResourcesContainer> _init_foreach_res_(
+    ConfigurationParser* my_config,
+    void* var)
+{
+  memset(var, 0, sizeof(void*)); 
+  return my_config->GetResourcesContainer();
+}
 // Loop through each resource of type, returning in var
 #ifdef HAVE_TYPEOF
```

Comment:
I guess TYPEOF is available in all compilers we use?

Reply:
After checking the config.h on all platforms I realized that HAVE_TYPEOF is not set on any of them. AFAICT the check for this is missing from CMake.

Reply:
After giving it a second and third thought we can probably just migrate to `decltype()` whereever typeof had been used.
I'll try to come up with a reasonable change for that.